mesonbuild/compilers/cpp.py

```diff
@@ -183,6 +183,14 @@ class _StdCPPLibMixin(CompilerMixinBase):
 
     """Detect whether to use libc++ or libstdc++."""
 
+    @functools.lru_cache(None)
```

Comment:
I think we don't need caching here, because has_header_symbol and has_header already will cache everything.

So caching with functools won't save us on forking out to the compiler, but does require associating an entire env object with the result answer.

Reply:
Quite right..